ocs_ci/ocs/ui/helpers_ui.py

```diff
@@ -95,6 +98,157 @@ def format_locator(locator, string_to_insert):
+    tls_server_name=None,
+):
+    """
+    Test for creation of storage class with or without encryption via UI
+
+    Args:
+            setup_ui: login function on conftest file
+            encryption (bool): By default set to False, should be True for Encryption enabled Storage Class
+            backend_path (str): name of the vault backend path
+            vault_namespace (str): name of the vault namespace
+            reclaim_policy (str): value of the reclaim policy, it could be 'Delete' or 'Retain'
+            provisioner (str): type of provisioner used, it could be 'rbd' or 'cephfs'
+            vol_binding_mode (str): value of the volume binding mode, it could be 'WaitForFirstConsumer' or 'Immediate'
+            service_name (str): the default value is None which can be changed in the function call
+            kms_address (str): the default value is None which can be changed in the function call
+            tls_server_name (str): the default value is None which can be changed in the function call
+
+    Returns:
+            sc_name (str) if the storage class creation is successful, returns False otherwise
+    """
+    base_ui_obj = PageNavigator(setup_ui)
+
+    ocp_version = get_ocp_version()
+    pvc_loc = locators[ocp_version]["storage_class"]
+
+    base_ui_obj.navigate_storageclasses_page()
+    logger.info("Create Storage Class")
+    base_ui_obj.do_click(pvc_loc["create-sc"])
+    logger.info("Storage Class Name")
+    sc_name = create_unique_resource_name(
+        resource_description="test", resource_type="storageclass"
+    )
+    base_ui_obj.do_send_keys(pvc_loc["sc-name"], f"{sc_name}")
+    logger.info("Storage Class Description")
+    base_ui_obj.do_send_keys(pvc_loc["sc-description"], "this is a test storage class")
+    logger.info("Storage Class Reclaim Policy")
+    base_ui_obj.do_click(pvc_loc["reclaim-policy"])
+    if reclaim_policy == "Delete":
+        base_ui_obj.do_click(pvc_loc["reclaim-policy-delete"])
+    elif reclaim_policy == "Retain":
+        base_ui_obj.do_click(pvc_loc["reclaim-policy-retain"])
+
+    if ocp_version >= "4.9":
+        logger.info("Volume binding mode")
+        base_ui_obj.do_click(pvc_loc["volume_binding_mode"])
+        if vol_binding_mode == "WaitForFirstConsumer":
+            logger.info("select WaitForFirstConsumer")
+            base_ui_obj.do_click(pvc_loc["wait_for_first_consumer"])
+        elif vol_binding_mode == "Immediate":
+            logger.info("select Immediate")
+            base_ui_obj.do_click(pvc_loc["immediate"])
+
+    logger.info("Storage Class Provisioner")
+    base_ui_obj.do_click(pvc_loc["provisioner"])
+    if provisioner == "rbd":
+        base_ui_obj.do_click(pvc_loc["rbd-provisioner"])
+    elif provisioner == "cephfs":
+        base_ui_obj.do_click(pvc_loc["cephfs-provisioner"])
+    logger.info("Storage Class Storage Pool")
+    base_ui_obj.do_click(pvc_loc["storage-pool"])
+    base_ui_obj.do_click(pvc_loc["ceph-block-pool"])
```

Comment:
It would be good to also have the option to create a new CBP, but that can be taken up as part of a different PR.